crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -0,0 +1,88 @@
+use std::{future::Future, pin::Pin, sync::Arc};
+
+use jsonrpsee::core::RpcResult;
+use reth_primitives::{rpc::Filter, U256};
+use reth_provider::BlockProvider;
+use reth_rpc_api::EthFilterApiServer;
+use reth_rpc_types::{FilterChanges, Index, Log};
+use reth_transaction_pool::TransactionPool;
+
+/// `Eth` filter RPC implementation.
+#[derive(Debug, Clone)]
+pub struct EthFilter<Pool, Client> {
+    /// All nested fields bundled together.
+    inner: Arc<EthFilterInner<Pool, Client>>,
+}
+
+impl<Pool, Client> EthFilter<Pool, Client> {
+    /// Creates a new, shareable instance.
+    pub fn new(client: Arc<Client>, pool: Pool) -> Self {
+        let inner = EthFilterInner { client, pool };
+        Self { inner: Arc::new(inner) }
+    }
+}
+
+impl<Pool, Client> EthFilterApiServer for EthFilter<Pool, Client>
+where
+    Pool: TransactionPool + 'static,
+    Client: BlockProvider + 'static,
+{
+    fn new_filter(&self, _filter: Filter) -> RpcResult<U256> {
+        todo!()
+    }
+
+    fn new_block_filter(&self) -> RpcResult<U256> {
+        todo!()
+    }
+
+    fn new_pending_transaction_filter(&self) -> RpcResult<U256> {
+        todo!()
+    }
+
+    fn filter_changes<'life0, 'async_trait>(
+        &'life0 self,
+        _index: Index,
+    ) -> Pin<Box<dyn Future<Output = RpcResult<FilterChanges>> + Send + 'async_trait>>
+    where
+        'life0: 'async_trait,
+        Self: 'async_trait,
+    {
+        todo!()
+    }
+
+    fn filter_logs<'life0, 'async_trait>(
+        &'life0 self,
+        _index: Index,
+    ) -> Pin<Box<dyn Future<Output = RpcResult<Vec<Log>>> + Send + 'async_trait>>
```

Comment:
this is the output `async_trait` generics, you can use the same `async` function the trait uses if you annotate this trait impl with #[async_trait] as well.